transport/conn.go

```diff
@@ -437,3 +438,19 @@ func (c *Conn) Close() {
 func (c *Conn) String() string {
 	return fmt.Sprintf("[addr=%s shard=%d]", c.conn.RemoteAddr(), c.shard)
 }
+
+func (c *Conn) registerEvents(e []frame.EventType) (responseHandler, error) {
+	h := make(responseHandler, 2)
+	c.r.h[eventStreamID] = h
```

Comment:
Although there is only one writer to this connection, I would do it under mutex to keep it clean. What do you think?

Reply:
If we want to do it with mutex (which is probably not necessary, but won't effect efficiency), we should wrap it in some method. I will try to implement this approach.

Reply:
thanks!